tangelo/toolboxes/circuits/lcu.py

```diff
@@ -381,7 +382,7 @@ def get_lcu_qubit_op_info(qu_op: QubitOperator()) -> Tuple[List[int], List[int],
     """
 
     max_qu_op = count_qubits(qu_op)
-    num_terms = len(qu_op.terms)
+    num_terms = len([coeff for coeff in qu_op.terms.values() if abs(coeff) > 1.e-8])
```

Comment:
If some coefficients are very close to 0, `num_terms` will then not match `len(qu_op.terms)`, which seems a bit misleading to me, and can cause users to have mismatch and unexpected differences when manipulating term-related info. If these terms "do not matter", how about we just remove them ?

I think maybe a good approach could be to set a parameter `threshold` with default value `1.e-8`, that the user can override, and anything below the threshold in absolute value vanishes entirely. 

I don't want to make this a request for change without getting clarity from you, but I'm not ready to approve this either.